sm/tedge_agent/tests/main.rs

```diff
@@ -0,0 +1,39 @@
+#[cfg(test)]
+#[cfg(feature = "mosquitto-available")]
```

Comment:
To be used as cfg flag the feature needs to be declared in the Cargo.toml.

But I would simply get rid of the conditional compilation here: the test should work even if the agent doesn't properly connect to mosquitto.

Reply:
I think there was a test breaking if these lines were, removed